src/Adapter/LegacyContextLoader.php

```diff
@@ -118,8 +118,14 @@ public function loadControllerContext(?string $controllerClassName = null): self
      */
     public function loadCurrencyContext(?int $currencyId = null): self
     {
-        $currency = new Currency($currencyId);
+        $useDefaultCurrency = false;
         if (null === $currencyId) {
+            $useDefaultCurrency = true;
+            $currencyId = Currency::getDefaultCurrencyId();
+        }
+
+        $currency = new Currency($currencyId);
+        if ($useDefaultCurrency) {
```

Comment:
```suggestion
        if (null === $currencyId) {
            $currency = new Currency(Currency::getDefaultCurrencyId());
            $currency->precision = Precision::DEFAULT_PRECISION;
        } else {
            $currency = new Currency($currencyId);
        }
```
This can maybe be simplified